### PR TITLE
gcc: update to 9.1.0

### PR DESCRIPTION
**Pull request by pullmoll**
Remove `libmpx{,-devel}` as they are no longer supported.

No longer try to build `gcc-gdc` as cross compiling fails and I don't feel like wasting more time on it.

Closes: #11663

[ci skip]

### CODE REVIEW

**Comment by xtraeme** [comment]
I see that cross compilers don't need gcc-ada anymore, why's that?

**Comment by pullmoll** [comment]
@xtraeme It's in base anyway so no need to explicitly list the dependency.

**Comment by xtraeme** [comment]
@pullmoll ok, I thought so.

**Comment by pullmoll** [comment]
Last thing to think about: perhaps we should set `gcc` to `nocross="Wait until cross compilers are ready"` to have `gcc-9.1.0` later built with itself. Not that it should make a difference, because we always bootstrap gcc, but I wonder if the cross builders will start to build gcc for the native architecture first?

**Comment by xtraeme** [comment]
@pullmoll yes please do. the cross builders will try to build the host pkg if it's not found. Better to make them wait until they are ready.

**Comment by xtraeme** [comment]
@pullmoll about libmpx, we should add the `replaces` for users that got it installed.

**Comment by protonesso** [comment]
mpx removed since gcc 9. So add it in conflicts to remove it

**Comment by pullmoll** [comment]
Hmm.. conflicts? No. I think `replaces="libmpx>=0 libmpx-devel>=0"` in the template's main section would be correct?

**Comment by xtraeme** [comment]
Yes, `replaces` is ok, don't use `conflicts`.

**Comment by protonesso** [comment]
xbps doesn't support conflicts removal before update?

**Comment by pullmoll** [comment]
Oops.. libmtx not libmpx

File: srcpkgs/cross-x86_64-w64-mingw32/template [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+_isl_version=0.19
+_mingw_version=6.0.0
 
 pkgname=cross-x86_64-w64-mingw32
-version="$_mingw_version"
-revision=1
-archs="x86_64* i686*"
+version=$_mingw_version
+revision=2
+archs="x86_64 i686"
```
Comment (xtraeme) [inline srcpkgs/cross-x86_64-w64-mingw32/template]: why is this restricted to glibc again?

**Comment by xtraeme** [comment]
@protonesso this is what `replaces` is for.

**Comment by protonesso** [comment]
Oh okay

File: srcpkgs/cross-x86_64-w64-mingw32/template (hunk 1 above)
Comment (pullmoll) [reply srcpkgs/cross-x86_64-w64-mingw32/template]: Seems to be due to me doing a wrong upstream merge at some point.

Comment (pullmoll) [reply srcpkgs/cross-x86_64-w64-mingw32/template]: fixed

**Comment by pullmoll** [comment]
Added a replaces for `libmpx-devel-32bit` in `gcc-multilib` as well.

**Comment by Johnnynator** [comment]
@pullmoll Can you regenerate the package signatures of your repo at https://repo.voidlinux.de/gcc91/ ? There are a few missing and quite a few are outdated.
I wanted to check if we hit https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90949 with telegram-desktop (which we should be affected)

**Comment by pullmoll** [comment]
I'll purge the signatures and redo them.

**Comment by Johnnynator** [comment]
> Also built `telegram-desktop` for `x86_64` now.

Yeah, I already did that myself, as expected it is segfaulting.

**Comment by pullmoll** [comment]
In gcc-6.x we had to set some flags for qt5 and IIRC it was `-fdisable-null-pointer-checks` or similar. Perhaps this helps here as well.

**Comment by xtraeme** [comment]
@Johnnynator would be nice to know the result from gdb.

**Comment by Johnnynator** [comment]
> @Johnnynator would be nice to know the result from gdb.

https://bugzilla.redhat.com/show_bug.cgi?id=1708621

gcc optimzes this line: https://github.com/telegramdesktop/tdesktop/blob/8af684f7f78f6c317323e1d1f2829e90e3bbd580/Telegram/SourceFiles/data/data_session.cpp#L3304
away

**Comment by pullmoll** [comment]
I'll build a gcc with the PR90949 patches applied and upload it to the repo once it's done.

**Comment by Johnnynator** [comment]
Yep, `-fno-delete-null-pointer-checks` is unrelated.

**Comment by pullmoll** [comment]
Rebuilt `telegram-desktop` with the patched `gcc` now. Can you test @Johnnynator ?

**Comment by pullmoll** [comment]
FWIW For me `Telegram` starts in a gcc91 environment but I don't have a mobile number to test any further...

**Comment by Johnnynator** [comment]
> FWIW For me `Telegram` starts in a gcc91 environment but I don't have a mobile number to test any further...

The crash happens later in the code when you are logged in, and yes the patch for gcc did fix it.

**Comment by xtraeme** [comment]
@pullmoll this should be ready to merge now?

**Comment by pullmoll** [comment]
I think so. I've updated the live ISOs at https://repo.voidlinux.de/gcc91/live/ again just in case.

**Comment by xtraeme** [comment]
I just tried the enlightenment image and seems to work just fine.

Please merge this ASAP. I'm excited with gcc-9.1 after reading the release notes.

Cheers

**Comment by Geod24** [comment]
> No longer try to build `gcc-gdc` as cross compiling fails and I don't feel like wasting more time on it.

@pullmoll : I came across this PR (and the linked issue) while working on supporting GDC on Alpine Linux. Had to cherry-pick a rather large diff, and make a few upstream fixes, but it is now working for me. I don't use Void, but I figured you or someone else might be interested: https://github.com/alpinelinux/aports/pull/12006

**Pull request merged**